snuba/clickhouse/translators/snuba/mappers.py

```diff
@@ -0,0 +1,79 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from snuba.clickhouse.translators.snuba import SnubaClickhouseStrictTranslator
+from snuba.clickhouse.translators.snuba.allowed import (
+    ColumnMapper,
+    SubscriptableReferenceMapper,
+)
+from snuba.query.dsl import array_element
+from snuba.query.expressions import Column, FunctionCall, SubscriptableReference
+
+
+@dataclass(frozen=True)
+class SimpleColumnMapper(ColumnMapper):
+    """
+    Maps a column with a name and a table into a column with a different name and table.
+
+    The alias is not transformed.
+    """
+
+    from_col_name: str
+    from_table_name: Optional[str]
+    to_col_name: str
+    to_table_name: Optional[str]
```

Comment:
> Reading this really makes me wish that the alias was separate from the Column type

We did that in the first iteration of the AST, early in the review. If I remember ther reason why we reverted to alias being an attribute was that all expression can have one in clickhouse. 
I think something like this would be a viable change.
```
class AliasedExpression(Expression):
    alias: str
    expression: Expression
```
It would make it harder to mess up though being aliases expressions themselves, they would be still present in the AST and thus translatable and transformable.

Though I am not sure I get what the connection of the rest of the comment is with aliases.

Reply:
> Though I am not sure I get what the connection of the rest of the comment is with aliases.

Yeah, sorry — it was a pretty bad description and kind of intertwined two somewhat related trains of thought. There were two points here:

1. It seems that there would be two instance attributes in this class in an ideal scenario — `to_column` and `from_column` (ignoring reserved words), both `Columns`, and that the `if` in `attempt_map` would be better expressed as `if expression == self.from_column`. Since the `Column` has the alias on it already, this doesn't really work.
2. If the `path` attribute gets added to `Column` (and no changes to aliasing), it seems like that logic should get also implemented here for consistency — or it might imply that the path is something separate from the `Column` itself, if that equality relationship above doesn't make sense, broadly speaking.

I think I'd be okay with (prefer, really) only allowing aliases in the `SELECT` clause at this point, though I'm not sure if there's anything that happens already that depends on aliasing being available elsewhere (I hope not.)

Reply:
> I think I'd be okay with (prefer, really) only allowing aliases in the SELECT clause at this point, though I'm not sure if there's anything that happens already that depends on aliasing being available elsewhere (I hope not.)

Guess what? yes there is :( 
And guess what? It is related to tags_key/tags_value. (from very early on, cannot say when this happened).
What happens is that the code that generates the old style arrayJoin operation,  does the join only once with keys and value, produces a bidimensional array, adds an alias to that and then refer to that alias to fetch one column or the other depending if the user wants keys or values 
https://github.com/getsentry/snuba/blob/master/snuba/datasets/tags_column_processor.py#L309
So internally we do use them to avoid (supposedly) calculating twice the same expensive operation.

I don't think the fact that we restricted aliases to the select expression should be a limiting factor to revisit this. I think what we should gauge carefully for the impact on query processing. The downside here is that this mapper should be in sync with the column structure, the downside of having aliases being nodes of the ast themselves means they would have to be taken into account during query processing (like always checking if the expression is a column or a column nested in an alias). This can be mitigated with the pattern system. 
I think it is something to explore.

About this specific mapper, I meant it more as a showcase to illustrate how the translator works. I think there are several possible alternatives on how to map columns. Ideally I would like this to look like more a dictionary where you provide references to the columns from a schema and they are mapped into columns of another schema without caring about each field of the column.